core/ui/src/views/settings/SettingsSmartHost.vue

```diff
@@ -94,45 +94,46 @@
                   <span v-html="$t('smarthost.port_label_tooltip')"></span>
                 </template>
               </NsTextInput>
-              <cv-toggle
-                :label="$t('smarthost.tls')"
-                value="statusValue"
-                :form-item="true"
-                v-model="tls"
+              <cv-combo-box
+                v-model="encrypt_smtp"
+                :label="$t('smarthost.choose')"
+                :title="$t('smarthost.encrypt_smtp')"
+                :auto-filter="true"
+                :auto-highlight="true"
+                :options="options"
                 :disabled="loading.getSmarthost || loading.setSmarthost"
+                :invalid-message="error.encrypt_smtp"
+                light
+                ref="encrypt_smtp"
               >
-                <template slot="text-left">
-                  {{ $t("common.disabled") }}
-                </template>
-                <template slot="text-right">
-                  {{ $t("common.enabled") }}
-                </template>
-              </cv-toggle>
-              <cv-toggle
-                :label="$t('smarthost.tls_verify')"
-                value="statusValue"
-                :form-item="true"
-                v-model="tls_verify"
-                :disabled="loading.getSmarthost || loading.setSmarthost"
-              >
-                <template slot="text-left">
-                  {{ $t("common.disabled") }}
-                </template>
-                <template slot="text-right">
-                  {{ $t("common.enabled") }}
-                </template>
-              </cv-toggle>
+              </cv-combo-box>
+              <template v-if="encrypt_smtp != 'none'">
+                <cv-toggle
+                  :label="$t('smarthost.tls_verify')"
+                  value="statusValue"
+                  :form-item="true"
+                  v-model="tls_verify"
+                  :disabled="loading.getSmarthost || loading.setSmarthost"
+                >
+                  <template slot="text-left">
+                    {{ $t("common.disabled") }}
+                  </template>
+                  <template slot="text-right">
+                    {{ $t("common.enabled") }}
+                  </template>
+                </cv-toggle>
+              </template>
               <div ref="setSmarthostError">
-                <div v-if="error.test_smarthost" class="bx--row">
-                  <div class="bx--col">
+                <cv-row v-if="error.test_smarthost">
+                  <cv-column>
                     <NsInlineNotification
                       kind="error"
                       :title="$t('action.set-smarthost')"
                       :description="error.test_smarthost"
```

Comment:
```suggestion
                      :description="error.setSmarthost"
```

Reply:
Is `setSmarthostAborted()` method missing?

Reply:
not sure to understand, or maybe I did a hack :-?

test_smarthost is sent by the API if I cannot authenticate

Reply:
L301 I set to false before to test again

Reply:
- What's the difference between `error.setSmarthost` and `error.test_smarthost`? Do we need both? https://github.com/NethServer/ns8-core/blob/6da11fe91427261d6a0c46605b9502881c21871d/core/ui/src/views/settings/SettingsSmartHost.vue#L300-L301
- `setSmarthostAborted()` method is used but doesn't exists.. I think we need to implement it https://github.com/NethServer/ns8-core/blob/6da11fe91427261d6a0c46605b9502881c21871d/core/ui/src/views/settings/SettingsSmartHost.vue#L313

Reply:
setSmarthost is a specific error for each parameter field (eg test if the regex none/starttls/tls is good
test_smarthost is a combination of all parameters to check if we could authenticate to the SMTP server (do we have the good expected encryption, the good password, the good TLS verified certificate....)

Reply:
good catch for setSmarthostAborted()